### PR TITLE
simgrid: add variant and remove flag

### PR DESCRIPTION
**Pull request by llnns**
Updates package simgrid:
- add a new variant for the old msg interface
- remove a flag from the compiler path

### CODE REVIEW

File: var/spack/repos/builtin/packages/simgrid/package.py [hunk 1]
```diff
@@ -132,7 +133,7 @@ def setup_dependent_package(self, module, dep_spec):
 
         if self.spec.satisfies("+smpi"):
             self.spec.smpicc = join_path(self.prefix.bin, "smpicc")
-            self.spec.smpicxx = join_path(self.prefix.bin, "smpicxx -std=c++11")
+            self.spec.smpicxx = join_path(self.prefix.bin, "smpicxx")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/simgrid/package.py:136]: Great catch.

File: var/spack/repos/builtin/packages/simgrid/package.py [hunk 2]
```diff
@@ -145,7 +146,8 @@ def cmake_args(self):
             args.append("-Denable_documentation=OFF")
         if spec.satisfies("+mc"):
             args.append("-Denable_model-checking=ON")
-
+        if spec.satisfies("+msg"):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/simgrid/package.py:149]: I realize the pattern is already established in this package, but did you know there are built-in methods that can help set `cmake` arguments from variants?  Check out https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/simgrid/package.py:149]: You can see more on `define` and `define_from_variant` starting at https://spack.readthedocs.io/en/latest/spack.build_systems.html?highlight=cmakepackage#spack.build_systems.cmake.CMakePackage.define.

**Review by tldahlgren — APPROVED** [review]
LGTM. Will defer merge decision until maintainer has a chance to comment.

**Review by viniciusvgp — APPROVED** [review]

**Comment by llnns** [comment]
Hello, maybe the pipeline needs to run again?

**Review by viniciusvgp — APPROVED** [review]

**Comment by tldahlgren** [comment]
@spackbot run pipeline

**Pull request merged**